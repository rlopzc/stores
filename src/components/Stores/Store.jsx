import React from 'react';

const Store = (props) => (
  <li>{props.name}</li>
);

export default Store;
