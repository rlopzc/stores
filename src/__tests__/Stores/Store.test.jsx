import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Store from '../../components/Stores/Store';


describe('Stores/Store', () => {
  const store = {position: {lat: 1, lng: 2}, name: 'store'};

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Store {...store}/>, div);

    const component = renderer.create(<Store {...store} />);
    expect(component).toMatchSnapshot();
  });

  it('renders an item with the store name', () => {
    const component = shallow(<Store {...store} />);
    const expected = store.name;
    expect(component.find('li').text()).toBe(expected);
  });
});
