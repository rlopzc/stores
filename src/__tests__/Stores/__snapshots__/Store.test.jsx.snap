// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stores/Store renders without crashing 1`] = `
<li>
  store
</li>
`;
